test/wallet-records-test.js

```diff
@@ -0,0 +1,425 @@
+    });
+
+    it('should return block hash', () => {
+      const data = getRandomBlockMetaData();
+      const meta = new BlockMeta();
+      meta.inject(data);
+
+      assert.bufferEqual(meta.toHash(), data.hash);
+    });
+
+    it('should be created from ChainEntry', () => {
+      const data = getRandomBlockMetaData();
+      const entry = new ChainEntry();
+      entry.hash = data.hash;
+      entry.height = data.height;
+      entry.time = data.time;
+
+      const meta = BlockMeta.fromEntry(entry);
+      compareBlockMeta(meta, data);
+    });
+  });
+
+  describe('TXRecord', function() {
+    const emptyBlock = {
+      height: -1,
+      block: null,
+      time: 0
+    };
+
+    const emptyTX = {
+      tx: null,
+      hash: null
+    };
+
+    it('should initialize with defaults', () => {
+      const wtx = new TXRecord();
+      const mtime = wtx.mtime;
+
+      compareTXRecord(wtx, {
+        mtime: mtime,
+        index: -1,
+        ...emptyTX,
+        ...emptyBlock
+      });
+    });
+
+    it('should initialize w/ tx', () => {
+      const {data, tx} = getRandomTXRecordData(true);
+      const wtx = new TXRecord(tx);
+
+      compareTXRecord(wtx, data);
+    });
+
+    it('should initialize w/ tx and block', () => {
+      const {data, block, tx} = getRandomTXRecordData(true, true);
+      const wtx = new TXRecord(tx, block);
+
+      compareTXRecord(wtx, data);
+    });
+
+    it('should fail encode w/o tx', () => {
+      let err;
+      try {
+        const wtx = new TXRecord();
+        wtx.encode();
+      } catch (e) {
+        err = e;
+      }
+
+      assert(err, 'Should throw error w/o tx');
+    });
+
+    it('should encode/decode w/ tx', () => {
+      const {data, tx} = getRandomTXRecordData(true);
+
+      const wtx = new TXRecord(tx);
+      const encoded = wtx.encode();
+      const decoded = TXRecord.decode(encoded);
+
+      compareTXRecord(wtx, data);
+      compareTXRecord(decoded, data);
+    });
+
+    it('should encode/decode w/ tx and block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+
+      const wtx = new TXRecord(tx, block);
+      const encoded = wtx.encode();
+      const decoded = TXRecord.decode(encoded);
+
+      compareTXRecord(wtx, data);
+      compareTXRecord(decoded, data);
+    });
+
+    it('should initialize from TX', () => {
+      const {data, tx} = getRandomTXRecordData(true);
+      const wtx = TXRecord.fromTX(tx);
+      compareTXRecord(wtx, data);
+    });
+
+    it('should initialize from TX and Block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+      const wtx = TXRecord.fromTX(tx, block);
+      compareTXRecord(wtx, data);
+    });
+
+    it('should set and unset block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+      const wtx = TXRecord.fromTX(tx);
+
+      assert.strictEqual(wtx.getBlock(), null);
+      assert.strictEqual(wtx.getDepth(random.randomInt()), 0);
```

Comment:
Why'd you test with random here? It'll always be `0` right? Just a bit unusual for our test style.

Reply:
The main purpose of these tests is to make sure serializations are correct, they don't correspond to the logic they are being used. So I think, random data better tests serialization than just using fixed value.

Reply:
OK